src/main/java/kai/task/TaskList.java

```diff
@@ -0,0 +1,117 @@
+package kai.task;
+
+import java.util.List;
+import java.util.ArrayList;
+
+public class TaskList {
+    private List<Task> tasks;
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Adds task to the list.
+     *
+     * @param task
+     */
+    public String addTask(Task task) {
+        String result;
+        tasks.add(task);
+        result = "Got it. I've added this task:\n"
+                + task + "\n"
+                + "Now you have " + getSize()
+                + " tasks in the list.";
+        return result;
+    }
+
+    /**
+     * Get size of the list.
+     *
+     * @return List size.
+     */
+    public int getSize() {
+        return tasks.size();
+    }
+
+    /**
+     * Lists down tasks in order when command is called.
+     */
+    public String listDownTask() {
+        String result;
+        int size = getSize();
+        result = "Here are the tasks in your list:\n";
+        for(int i = 1; i <= size; i++) {
+            result += i +": " + tasks.get(i - 1) + "\n";
+        }
+        return result;
+    }
+
+    /**
+     * Deletes the task at that index from the list.
+     *
+     * @param index
+     */
+    public String deleteTask(int index) {
+        String result;
+        Task removed = getTask(index - 1);
+        result = "Noted. I've removed this task:\n" + removed + "\n";
+        tasks.remove(index - 1);
+        result += "Now you have " + getSize() + " tasks in the list";
+        return result;
+    }
+
+    /**
+     * Change the specific task at index as done.
+     *
+     * @param index
+     */
+    public String markTask(int index) {
```

Comment:
Could further specify what exactly is being marked i.e. markTaskAsDone for better readability